src/documentation/programming-guide.md

```diff
@@ -287,18 +368,33 @@ static class ComputeWordLengthFn extends DoFn<String, Integer> {
 }
 ```
 
+```python
+class ComputeWordLengthFn(beam.DoFn):
+  def process(self, context):
+    # Get the input element from ProcessContext.
+    word = context.element
+    # Use return to emit the output element.
+    return [len(word)]
+```
+
+{:.language-java}
 > **Note:** If the elements in your input `PCollection` are key/value pairs, you can access the key or value by using `ProcessContext.element().getKey()` or `ProcessContext.element().getValue()`, respectively.
 
-A given `DoFn` instance generally gets invoked one or more times to process some arbitrary bundle of elements. However, Beam doesn't guarantee an exact number of invocations; it may be invoked multiple times on a given worker node to account for failures and retries. As such, you can cache information across multiple calls to your `@ProcessElement` method, but if you do so, make sure the implementation **does not depend on the number of invocations**.
+{:.language-python}
+> **Note:** If the elements in your input `PCollection` are key/value pairs, you can access the key or value by using `[key for key in context.element]` or `[context.element[key] for key in context.element]`, respectively.
```

Comment:
Python SDK uses tuple() for key value pairs. If context element is of that form the following code would extract bits:

`key, value = context.element`

Since tuple() is a native element to Python, I do not think there is a need for this paragraph in Python.